### ISSUE TITLE
grabInput permanently changes the behavior of the term emulator.

### ISSUE THREAD

**Issue by ashnur** [cronvel/terminal-kit#2]
After using `grabInput({mouse: 'button'})` the default behavior of the mouse events gets swallowed (this is expected) and each mouse action spits stuff into the term (e.g. `0;4;7M`, this is not expected and I do not know how to avoid or circumvent it), this doesn't stop even after `grabInput(false)` and `process.exit()`, making the term window useless after I ran the script.

I tried this with terminology and gnome-terminal.

**Comment by cronvel** [comment]
Hi, can you provide me some code so I can test that?
What is your gnome version? Your distro?

I'm afraid, it works fine on mine.

You can test using the program: sample/all-event-test.js,
Also try using Ctrl-D during the test to switch from one mouse reporting to another.

**Comment by ashnur** [comment]
I tried all-event-test, it was working better than what I tried before, because there were no issues until I got to `motion`. But then this happened: 
![2015-05-28-165441_1920x1080_scrot](https://cloud.githubusercontent.com/assets/407934/7861704/68ec8f3e-055a-11e5-84d8-70eca10e1326.png)

I only have part of gnome installed (so no overall gnome version), but my gnome-terminal is 3.16.2-1 
I am using ArchLinux.

However, given the fact that your test is clearly working for me (at least the button and drag modes, which I need), I started thinking what else did I differently. Turns out, using nodemon causes the issue I was talking about. If I don't use nodemon, the issues go away.

Now, I am not sure if this is a problem with terminal-kit or nodemon or neither, just bad luck, but I think we can close this issue for now.
Thanks for the help :).

**Comment by cronvel** [comment]
There are some chances that nodemon does not pass STDIN and STDOUT data as they are (i.e. performs some formatting, or spawn its own PTY, or whatever), thus interferes with escape sequences written back and forth.

However your screenshot is clearly a bug I should squash ;)

**Comment by ashnur** [comment]
Tell me if I can help you with that.

**Comment by cronvel** [comment]
I can't reproduce it but I made some improvement to avoid the whole app to crash if that happens again.
If you can reproduce it again, it should give us more informations.

**Comment by ashnur** [comment]
I did a pull and tried all-events again. Now, it didn't crash and quit the process, it just shows this:
![2015-05-28-192705_1920x1080_scrot](https://cloud.githubusercontent.com/assets/407934/7865203/bd099674-056f-11e5-9985-29f05c9fe1f7.png)

**Comment by ashnur** [comment]
And in gnome-terminal, it doesn't happen, there it works. So it's probably terminology being broken in some way, if you could tell me what (as in, what standard or something) is not implemented or broken, I could make a request to the authors of terminology.

**Comment by cronvel** [comment]
Ok, I got it. The *minus* was ruining the parser. I don't know why they report a negative code here, it is supposed to be an unsigned integer. However, there isn't any real spec for mouse reporting, even [invisible island](http://invisible-island.net/xterm/ctlseqs/ctlseqs.html) do not give precise pointer.

I fixed that, assuming *-1* is used by Terminology instead of the more common code *32* for mouse motion.

Now it should work accordingly.

Thank you for helping me fixing bug!
